### PR TITLE
Add ``join_items`` to itertools.

### PR DESCRIPTION
**Pull request by jaheba**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: src/gluonts/itertools.py [hunk 1]
```diff
@@ -452,3 +452,31 @@ def power_set(iterable):
     return itertools.chain.from_iterable(
         itertools.combinations(iterable, r) for r in range(len(iterable) + 1)
     )
+
+
+def zip_items(*dicts, default=None, strict=False):
```
Comment (lostella) [inline src/gluonts/itertools.py]: zip_values seems more what is happening here?

File: src/gluonts/itertools.py [hunk 2]
```diff
@@ -452,3 +452,31 @@ def power_set(iterable):
...
+    """
+    Iterate over multiple dictionary items.
+
+    >>> a = {"x": "a"}
+    >>> b = {"x": "b"}
+    >>> c = {}
+    >>> for name, (xa, xb, xc) in zip_items(a, b, c):
+    ...     print(name, xa, xb, xc)
+    x a b None
```
Comment (lostella) [inline src/gluonts/itertools.py]: This contrasts with what happens for lists with zip, in which case the output is limited to the common keys. So in this example the output iterator would be empty.

Maybe adding None could be optional?

File: src/gluonts/itertools.py (hunk 1 above)
Comment (jaheba) [reply src/gluonts/itertools.py]: It returns key value pairs like `items`, hence the name.

File: src/gluonts/itertools.py (hunk 2 above)
Comment (jaheba) [reply src/gluonts/itertools.py]: I guess there are generally three ways to handle this:

* be strict and require all dicts to have the same keys
* just iterate over the intersection of all keys
* iterate over the union of all keys, fall back on a default when a dict doesn't have a given key

Would we want to support all three modes?

**Comment by jaheba** [comment]
I think the name is a bit confusing, maybe `join_items` or something like that is better?

**Comment by lostella** [comment]
Maybe yes; then one can borrow the SQL jargon and define inner, outer, left, and right to cover all cases

**Comment by jaheba** [comment]
`left` / `iight` only works for two inputs. I think that's fine?

**Review by lostella — APPROVED** [review]

**Pull request merged**